[Testing:Developer] Add Audit & Withdrawn to Sample Data

JManion32 opened the pull request:
fixes #11763

### Why is this Change Important & Necessary?
There is no way to test if TA grading statistics that omit withdrawn and auditing students from metrics on number of students submitted, number of submitted assignments that were manually graded, and average scores.

### What is the New Behavior?
Create 2 new users: one with `registration_type: audit` and another with `registration_type: withdrawn`

### What steps should a reviewer take to reproduce or test the bug or new feature?
Run `recreate_sample_course` -> see the new students in section 1 of each course -> verify that all other students have `registration_type: graded`

JManion32 committed:
Initial Commit

Set the randomized registration types in course_create

JManion32 committed:
Fixed Python Lint

JManion32 committed:
Scaling back the Percentages

Instead of 5 percent withdrawn and 5 percent audit, I think 2 percent for each would be more sufficient. This will also make integrating with Cypress tests easier.

jeffrey-cordero commented on .setup/bin/sample_courses/models/course/course_create.py:
```diff
@@ -26,6 +26,16 @@
 
 
+def pick_registration_type() -> str:
+    r = random.random()
```
This will lead to flaky e2e tests given the random nature of the registration type. Why don't we create a new user, `withdrawn`, in `.setup/data/users` and handle the create course script. Potentially `registration_section: 'withdrawn'` to set the `registration_type` in the current script?

JManion32 committed:
Update course_create.py

1 percent for each

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Added Users

Added a withdrawn, and an audit user. TODO: link with course_create.

registration_type is found in both courses_users and the users table in the course db, so I assume it can be set in a user.yml file, but am unsure. Regardless, I will need these users anyways, and I can change these attributes later

JManion32 committed:
Fully Revert course_create

JManion32 committed:
Add registration_type documentation

JManion32 committed:
Added Reg Type to Course Create

JManion32 committed:
Update course_create.py

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Update course_create.py

JManion32 committed:
Added registration_type to user __init__

registration_type can now be read from a user yml file

JManion32 committed:
Changed Registration Type

Should fail less Cypress now

JManion32 committed:
Update audit.yml

JManion32 committed:
Update withdrawn.yml

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Added another withdrawn and another audit

Added Scott Audit and Kaitlyn Withdrawn to section 4. Now we have 2 of each.

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Fixed YAML Lint

JManion32 committed:
Merge branch 'withdrawn-sample' of https://github.com/Submitty/Submitty into withdrawn-sample

JManion32 committed:
Updating Cypress Tests

Probably 40% complete.

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Fixed UI and Gradeable

JManion32 committed:
TAGrading and Feature

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Update navigation_page.spec.js

This is the second time I am rewriting this test and having it pass locally. I am worried that other tests aren't self-contained, thus causing this to fail on main despite working on a clean sample course locally.

JManion32 committed:
Update student_version_conflict.spec.js

JManion32 committed:
Update student_version_conflict.spec.js

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Possible good changes?

Idk this is almost impossible, I'm going to write up my struggles with each of the remaining tests and then move on. I have wasted far too much time trying to fix up this PR...

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Update grading_stats.spec.js

JManion32 committed:
Merge branch 'withdrawn-sample' of https://github.com/Submitty/Submitty into withdrawn-sample

JManion32 committed:
Update navigation_page.spec.js

JManion32 committed:
Fix version conflict, start rubric_grading

JManion32 committed:
Update navigation_page.spec.js

JManion32 committed:
Update rubric_grading.spec.js

JManion32 committed:
Revert "Update rubric_grading.spec.js"

This reverts commit f5477c97b432d782d7de6dbf2c88793879536dcd.

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Merge branch 'main' into withdrawn-sample

JManion32 committed:
Merge branch 'main' into withdrawn-sample

dandrecollins07-ctrl committed:
Merge branch 'main' into withdrawn-sample

dandrecollins07-ctrl reviewed (changes requested):
Hello, your code seems to be working. However they are various cypress issues. Once these are fixed, I can give a proper review and hopefully approve your PR.

dandrecollins07-ctrl committed:
Merge branch 'main' into withdrawn-sample
